include/RAJA/index/ListSegment.hpp

```diff
@@ -280,34 +387,68 @@ class TypedListSegment
   // Initialize segment data properly based on whether object
   // owns the index data.
   //
-  void initIndexData(const value_type* container,
+  void initIndexData(bool use_resource,
```

Comment:
Perhaps this should be protected/private since the user shouldn't be calling this

Reply:
It is private.